### PR TITLE
Info Window Clean Up

### LINKED ISSUE #21: Info window: Make selected item bigger than others

### LINKED ISSUE #27: Make Info Window dynamic
By adding a render_config with the fields to show on the info window

### LINKED ISSUE #29: Tests for Info Window Code
Add unit tests for the newly added code for the Info Window

### PR DESCRIPTION
**Pull request by jasonrogena**
Fixes:
 1. Making selected items on the info window bigger (#21)
 2. Dynamic info window items (#27)
 3. Tests for info window logic (#29)

Code has also been slightly moved around to make it better to understand.

Fixes #21 
Fixes #27 
Fixes #29

### CODE REVIEW

File: library/src/main/java/io/ona/kujaku/activities/MapActivity.java [hunk 1]
```diff
@@ -260,35 +288,26 @@ private LatLng getBoundsCenter(LatLng topLeftBound, LatLng bottomRightBound) {
...
-                bottomRightBound.getLatitude() + (latDifference/2),
-                topLeftBound.getLongitude() + (lngDifference/2)
+                bottomRightBound.getLatitude() + (latDifference / 2),
+                topLeftBound.getLongitude() + (lngDifference / 2)
         );
     }
 
     private void initializeViews() {
+        dismissAllDialogs();
+        alertDialogs = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (jasonrogena) [reply library/src/main/java/io/ona/kujaku/activities/MapActivity.java]: My assumption here is that initializeViews() is called whenever we want to redraw the UI (regardless of it being the first time the activity is loaded or not). You'd want all the alert dialogs to be dismissed when the UI is redrawn, I guess.

File: library/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,5 +1,8 @@
 <resources>
     <string name="app_name">Kujaku</string>
+    <string name="kujaku_config">Kujaku Configuration</string>
```
[the remark this answers is not in the document]
Comment (jasonrogena) [reply library/src/main/res/values/strings.xml]: The message doesn't say "Kujaku Configuration Error"...

File: sample/src/main/assets/2017-nov-27-kujaku-metadata.json [hunk 3]
```diff
@@ -3,18 +3,46 @@
   "name": "zeir-sample-1-kigamba",
   "metadata": {
     "kujaku": {
-      "data_source_names": [
-        "opensrp-custom-data-source-0",
-        "opensrp-custom-data-source-1",
-        "opensrp-custom-data-source-2",
-        "opensrp-custom-data-source-3"
+      "data_sources": [
```
[the remark this answers is not in the document]
Comment (jasonrogena) [reply sample/src/main/assets/2017-nov-27-kujaku-metadata.json]: OK

File: library/src/main/res/values/strings.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ekigamba) [reply library/src/main/res/values/strings.xml]: Ok then, the message that shows when there's an error just says Kujaku configuration

**Pull request merged**